syntax = "proto3";

package proseandbabel;

service ProseAndBabel {

rpc GetHaiku (BabelRequest) returns (Babel) {}
rpc GetBabel (BabelRequest) returns (Babel) {}
rpc UserMarkov (UserTweets) returns (UserBabel) {}
rpc UserHaiku (UserTweets) returns (UserBabel) {}
rpc GetFib (BabelRequest) returns (Babel) {}
rpc UserFib (UserTweets) returns (UserBabel) {}

rpc TestServerUp (Greeting) returns (Hello) {}
}

message BabelRequest{
  string ask = 1;
}

message Babel{
  string prose = 1;
}

message UserTweets{
  string tweets = 1;
}

message UserBabel{
  string prose = 1;
}


message Greeting{
  string greeting = 1;
}

message Hello{
  string hello = 1;
}
